### PR TITLE
[improve](cast) support ipv4 cast to ipv6 type

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

support ipv4 cast to ipv6 type

### CODE REVIEW

**Review by morrySnow — DISMISSED** [review]

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_cast.h [hunk 1]
```diff
@@ -1482,6 +1482,43 @@ class PreparedFunctionCast : public PreparedFunctionImpl {
...
+            const auto& ipv4_column_data = col_from_ipv4->get_data();
+
+            ColumnUInt8::MutablePtr col_null_map_to;
+            ColumnUInt8::Container* vec_null_map_to [[maybe_unused]] = nullptr;
+            col_null_map_to = ColumnUInt8::create(row, 0);
+            vec_null_map_to = &col_null_map_to->get_data();
+            for (size_t i = 0; i < row; ++i) {
+                map_ipv4_to_ipv6(ipv4_column_data[i], reinterpret_cast<UInt8*>(&vec_to[i]));
+            }
+            block.get_by_position(result).column =
```
Comment (HappenLee) [inline be/src/vec/functions/function_cast.h:1506]: why ipv4 cast to ipv6 always return nullable？

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**